x/gov/genesis_test.go

```diff
@@ -175,3 +175,48 @@ func TestEqualProposals(t *testing.T) {
 	require.Equal(t, state1, state2)
 	require.True(t, state1.Equal(state2))
 }
+
+func TestEqualProposalsV2(t *testing.T) {
+	app := simapp.Setup(t, false)
+	ctx := app.BaseApp.NewContext(false, tmproto.Header{})
+	addrs := simapp.AddTestAddrs(app, ctx, 2, valTokens)
+
+	SortAddresses(addrs)
+
+	header := tmproto.Header{Height: app.LastBlockHeight() + 1}
+	app.BeginBlock(abci.RequestBeginBlock{Header: header})
+
+	// Submit two proposals
+	govAccount := app.GovKeeper.GetGovernanceAccount(ctx)
+	// Proposal is for the gov module to vote on another proposal :)
+	proposalMsgs := []sdk.Msg{types.NewMsgVote(govAccount.GetAddress(), 0, types.OptionYes)}
+	proposal1, err := app.GovKeeper.SubmitProposalV2(ctx, proposalMsgs)
+	require.NoError(t, err)
+
+	proposal2, err := app.GovKeeper.SubmitProposalV2(ctx, proposalMsgs)
+	require.NoError(t, err)
+
+	// They are similar but their IDs should be different
+	require.NotEqual(t, proposal1, proposal2)
+	require.NotEqual(t, proposal1, proposal2)
```

Comment:
those 2 tests are the same

Reply:
Strange. I just copied and pasted the tests. We do the same behavior futher up